src/ServiceManagement/RecoveryServices/Commands.RecoveryServices/Service/InvokeAzureRecoveryServicesVaultUpgrade.cs

```diff
@@ -0,0 +1,238 @@
+                    this.WriteResponse(
+                        string.Format(
+                            Properties.Resources.VaultUpgradeSucceded,
+                            this.ResourceType,
+                            this.VaultName));
+                }
+                else
+                {
+                    this.WriteResponse(Properties.Resources.VaultUpgradeNotSucceded);
+                }
+            }
+            else if (this.StopProcessingFlag)
+            {
+                this.WriteResponse(Properties.Resources.VaultUpgradeTerminated);
+            }
+            else
+            {
+                this.WriteResponse(Properties.Resources.VaultUpgradeTimedOut);
+            }
+        }
+
+        /// <summary>
+        /// Validating vault upgrade prerequisites.
+        /// </summary>
+        /// <returns>CheckPrerequisites response.</returns> 
+        private CheckVaultUpgradePrerequisitesResponse ValidateVaultUpgradePrerequisites()
+        {
+            CheckVaultUpgradePrerequisitesResponse response =
+                CheckVaultUpgradePrerequisitesResponse.Succeeded;
+
+            this.WriteResponse(Properties.Resources.StartingPrerequisitesCheck);
+
+            try
+            {
+                this.RecoveryServicesClient.TestVaultUpgradePrerequistes(
+                    this.VaultName,
+                    this.Location,
+                    this.ResourceType,
+                    this.TargetResourceGroupName,
+                    this.Profile.Context.Subscription.Id.ToString());
+
+                this.WriteObject(Properties.Resources.CheckPrereqSucceded);
+            }
+            catch (Exception exception)
+            {
+                this.WriteResponse(Properties.Resources.CheckPrereqFailed);
+                ExceptionDetails details =
+                    this.RecoveryServicesClient.HandleVaultUpgradeException(exception);
+                if (!string.IsNullOrEmpty(details.WarningDetails))
+                {
+                    this.WriteWarning(details.WarningDetails);
+                }
+
+                if (!string.IsNullOrEmpty(details.ErrorDetails))
+                {
+                    throw new InvalidOperationException(
+                        Environment.NewLine +
+                        string.Format(
+                            Properties.Resources.ConfirmVaultUpgradePrereqFailed,
+                            Properties.Resources.VaultUpgradeExceptionDetails,
+                            details.ErrorDetails));
+                }
+
+                response = CheckVaultUpgradePrerequisitesResponse.SucceededWithWarnings;
+            }
+
+            return response;
+        }
+
+        /// <summary>
+        /// Writes content to the screen.
+        /// </summary>
+        /// <param name="contents">Data to be printed on the screen.</param>
+        private void WriteResponse(string contents)
+        {
+            this.WriteObject(Environment.NewLine);
```

Comment:
This method should not write to the Output stream - the Output stream contains the return value of the cmdlet.  This should write to the progress, debug, or verbose streams